### PR TITLE
Creating GitHubServiceSpi (issue #61)

### PR DESCRIPTION
**Pull request by rimolive**

### CODE REVIEW

File: services/github-service-impl/src/main/java/org/kontinuity/catapult/service/github/spi/GitHubServiceSpi.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        boolean has_issues, boolean has_wiki, boolean has_downloads) throws IOException, IllegalArgumentException;
+
+
+    /**
+     * Delete a repository specified by its value object representation.
+     *
+     * @param repository - the value objct the represents the Gitub repository
+     * @throws IllegalArgumentException
+     */
+    void deleteRepository(final GitHubRepository repository) throws IllegalArgumentException;
```
Comment (ALRubinger) [inline services/github-service-impl/src/main/java/org/kontinuity/catapult/service/github/spi/GitHubServiceSpi.java]: Needs some symmetry between `create` and `deleteRepository`, probably changing to `createRepository` as that's what `(create|delete)Webhook` looks like.

**Comment by ALRubinger** [comment]
Missing I think:

1) Move `create` from API to SPI (we need to remove it from API) 
2) Move `deleteWebhooks` from API to SPI

That plus the line note and we should be good I think :)

File: services/github-service-impl/src/main/java/org/kontinuity/catapult/service/github/spi/GitHubServiceSpi.java (hunk 1 above)
Comment (rimolive) [inline services/github-service-impl/src/main/java/org/kontinuity/catapult/service/github/spi/GitHubServiceSpi.java]: Agreed.

**Comment by ALRubinger** [comment]
c05d9f22716d3f30184cb24a116d545608b54165

**Pull request closed without merging**